### PR TITLE
Improve Coach tabs accessibility for quiz report page

### LINKED ISSUE #10253: Improve Coach tabs accessibility - "Report-Difficult questions" tabs on a Reports quiz page
In #10212, we embarked on the journey toward improved accessibility of the coach plugin. Part of it is refactoring tabs logic to use new KDS tabs components that are implemented in an accessible manner. This issue is a continuation of that work.

The goal is to use the KDS tabs components for _Report-Difficult questions_ tabs on a quiz page in the coach reports ( _Coach → Reports → Quizzes_ → select a quiz → quiz page)

report-difficultquestions

## Acceptance criteria

- [ ] The current implementation is refactored to use KDS tabs components (`KTabs` or the combination of `KTabsList` + `KTabsPanel`)
- [ ] The new implementation satisfies [these](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) and [these](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/) W3C guidelines (this is already provided by KDS tabs components, however in some cases, a bit of additional Kolibri logic may be needed as well - see comments below)
- [ ] The focus ring shows when navigating with a keyboard and stays on the selected tab after pressing enter

## Comments

- KDS tabs components documentation
  - [Tabs page](https://deploy-preview-426--kolibri-design-system.netlify.app/tabs/)
  - [KTabs component](https://deploy-preview-426--kolibri-design-system.netlify.app/ktabs/)
  - [KTabsList component](https://deploy-preview-426--kolibri-design-system.netlify.app/ktabslist/)
  - [KTabsPanel component](https://deploy-preview-426--kolibri-design-system.netlify.app/ktabspanel)
- You can play around with the _Lessons-Quizzes-Groups-Learners_ tabs that are already implemented with KDS components  to experience the expected behavior
example
- In some cases, due to the routing architecture, some additional Kolibri logic may be needed to ensure seamless focus behavior. If you run into issues with the focus ring disappearing after pressing enter on a tab or similar, [useCoachTabs composable](https://github.com/learningequality/kolibri/blob/5806f4ad50db3d96fdd0030d0659861b52a63e67/kolibri/plugins/coach/assets/src/composables/useCoachTabs.js) may help

### PR DESCRIPTION
**Pull request by KshitijThareja**
## Summary

This PR aims to improve the coach tabs accessibility for the quiz report page with the use of KDS tabs (KTabsList and KTabsPanel) in the `QUIZZES` tab which is inside Coach->Reports->Quizzes->Select a quiz. The new constants for the quiz tab are defined in `tabsConstants.js`.

https://github.com/learningequality/kolibri/assets/116485536/1595cff1-beb2-40dc-b9bc-641a462dc0a3

## References

Closes #10253 

## Reviewer guidance

1. Go to Coach->Reports-Quizzes-Select a quiz
2. For accessibility, use the keyboard and go to the **Report** tab and then press the right arrow to move focus to the **Difficult Questions** tab. Press enter to open the focused tab.
3. For normal, use the mouse to click the respective tab to open.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by KshitijThareja** [comment]
Hi @MisRob. I've made the PR targeting the `release-v0.16.x` branch, as discussed earlier. After the review is done, I can rebase it to the `develop` branch if required.
Thanks :)

**Comment by MisRob** [comment]
Thanks @KshitijThareja, great! I will review some time next week.

File: kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue [hunk 1]
```diff
@@ -113,6 +157,12 @@
         }
       },
     },
+    $trs: {
+      coachReportsQuizzes: {
```
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue]: @radinamatic This a string that's used as `aria-label` for this tablist

Screenshot from 2023-12-13 15-21-12

I was thinking if "Coach reports quiz tabs" would make more sense?

Also I think we could improve the string context to say something like "Labels the quiz page tabs in the Coach > Reports > Quizzes"

 I'd be grateful to hear from you before making a final suggestion to @KshitijThareja.

**Comment by MisRob** [comment]
Hi @KshitijThareja, thanks, great work. I haven't notice any issues in code and I manually tested it as well. All worked as expected for me. Just leaving one note about a string. I'd like to hear from @radinamatic before we make the final tweak. I think that we can target the release branch if our QA team can test it out, please :)? @radinamatic @pcenov

**Comment by KshitijThareja** [comment]
Thanks @MisRob :)
I'll be happy to work on any further suggestions by the team.

File: kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue (hunk 1 above)
Comment (radinamatic) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue]: Bigger issue here is that since the intention is to merge this into the 0.16 release, we cannot add new strings at this point, as there will be no translations for them available for this release.

Comment (radinamatic) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue]: Ideally the tablist aria-label should say something like ``Report details``, but we have nothing similar on Crowdin. Maybe just reusing the string ``Details`` that we do have in our code already, since this tablist is so deep into Coach and reports features that it would be specific enough.

Comment (MisRob) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue]: Ah okay, I didn't realize this. Thanks @radinamatic. Yes, then let's go with the existing `'Details'` string for now and I will open a follow-up issue for us to replace it with something better. @KshitijThareja The existing string is saved as `detailsLabel` defined in `commonCoachStrings.js`. Please use it instead of adding a new one. Thank you.

Comment (radinamatic) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue]: And just as a reminder: when writing strings to use in aria-labels for interactive elements, there is no need to include the type of the element as the screen reader already announces it. For example the tablist on the Coach home has the label ``Coach reports``, and the NVDA screen reader announces it as:

`Coach reports  tab control`

If you write the string for aria-label ``Coach reports quiz tabs`` as suggested above,  NVDA would announce it as:

`Coach reports quiz tabs tab control`

To sum it up, no need to put the element type inside its aria-label, it is redundant 🙂

Comment (MisRob) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue]: ❤️🙏🏻

**Comment by radinamatic** [comment]
And I missed to confirm that the keyboard navigability on this page is working as intended 🙂 
Good job, will approve! 👍🏽

**Comment by MisRob** [comment]
Great! @KshitijThareja so the only required update is to re-use the existing string. We will then proceed to merge. Good work!

**Review by radinamatic — APPROVED** [review]
Yey for more accessible tab controls in coach pages! 👏🏽 

As soon as the label is corrected, this will be good to merge! 💯

**Comment by KshitijThareja** [comment]
Thanks a lot @radinamatic @MisRob 😊
Will work on the suggested changes and update it as soon as possible.

File: kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue (hunk 1 above)
Comment (KshitijThareja) [reply kolibri/plugins/coach/assets/src/views/reports/ReportsQuizBaseListPage.vue]: I've changed the `aria-label` to use the `detailsLabel` from `commonCoachStrings.js` as asked 👍🏻

**Review by MisRob — APPROVED** [review]
Thanks @KshitijThareja and @radinamatic!
